### ISSUE TITLE
Deep merge operation took too much time

### ISSUE THREAD

**Issue by wshdota2** [reportportal/reportportal#2025]
If I want to deep merge the same name launch in the picture. It took me about 15 minutes. May be caused by too many cases in the launch. So I want to know how does deep merge work? and how to reduce deep merge time. Thanks a lot.

![e54a51d0084d2087c12c4e58fb954be](https://github.com/reportportal/reportportal/assets/51727190/7079e26c-81be-46cd-ab2b-c86db0975999)

**Comment by wshdota2** [comment]
![e54a51d0084d2087c12c4e58fb954be](https://github.com/reportportal/reportportal/assets/51727190/6dcc2251-acd0-4fb1-8de1-7e5a1228d83b)

**Comment by wshdota2** [comment]
I found that this step spend too much time, and the cpu not buzy at the same time.
![1685963547896](https://github.com/reportportal/reportportal/assets/51727190/91529be3-5afb-449c-acee-1e5eb8d00e42)

**Comment by irynakozak2** [comment]
@wshdota2 Hi! Unfortunately, we have a known performance issue with the Merge functionality, especially Deep. As far as the deep merge is required a lot of interactions with the PostgreSQL database to process such operation, my suggestion is to revise the configuration parameters for the PostgreSQL and setup them up according to [our recommendations](https://reportportal.io/docs/installation-steps/OptimalPerformanceHardwareSetup#5-postgresql-performance-tuning).

Appreciate your feedback and waiting for your reply.

**Comment by irynakozak2** [comment]
@wshdota2 Hi! Did you have a chance to review the proposed PostgreSQL configuration and apply it to your environment?

**Comment by wshdota2** [comment]
> @wshdota2 Hi! Did you have a chance to review the proposed PostgreSQL configuration and apply it to your environment?

Sorry for the late reply. I tried better parameters before raising this question, but the performance did not improve. I have summarized the following reasons:
1. There are too many layers in the testsuite, totaling 7 layers;
2. There are too many test cases, exceeding 30000;
After I adopted a 4-layer testsuite, the deep merge time was greatly improved. Thank you for your suggestion.
